node_ops/node_ops_ctl.hh

```diff
@@ -99,20 +99,23 @@ struct node_ops_cmd_request {
     std::unordered_map<gms::inet_address, std::list<dht::token>> bootstrap_nodes;
     // Optional field, list uuids of tables being repaired, set by repair cmd
     std::list<table_id> repair_tables;
+    std::unordered_map<gms::inet_address, locator::host_id> endpoint_to_host_id;
```

Comment:
Why is the mapping needed? Comment and commit message should explain.

Reply:
In the following commit it seems that in most places you're still using token_metadata on the handler side to do the host-ID -> IP mapping (`get_new()->get_host_id(node)`). So sending this mapping seems redundant. Although it could be potentially divergent from the remote's mapping... in which case I believe that the coordinator's mapping should indeed be the one we use, but then all those places where you do the mapping using local info should be modified to use the coordinator's mapping (i.e. the one I'm commenting on).

But this is only needed (is it?) in gossiper-mode topology operations and there, mappings need to be consistent anyway across nodes (by ring_delay waits between topology changes and IP changes), so it shouldn't matter which mapping we're using.

*If* we're still using this in raft-topology mode (I haven't checked), then topology operation requests might potentially race with IP changes. In that case we need to stick to a consistent mapping, which seems to be the node_ops coordinator's mapping. But then what if its mapping is outdated... I assume you thought about this for longer than I thought now when writing this comment, so you have some answers

BTW. would it work to have `node_ops_cmd_request_host_id` (or similar name), where `gms::inet_address` is replaced with `host_id`, and not depend on mappings at all?